### DISCUSSION TITLE
Oshavery v0.2.0にむけてのフロントエンドの環境再建

### DISCUSSION THREAD

**Discussion by re-taro** [Growthers/Oshavery#37] [Ideas]
# 要件
- Next.js(preact/react v17.0.2)
- Sass
- TypeScript
- Storybook
- Jest
- Conventional Commits
- なんちゃってgit flow

**Comment by re-taro** [comment]
# 議論点
- components/下のディレクトリ構成
- Style lintのルール
- ESLintのルール
- tsconfigの中身
- Prettierのルール
- commit lintを導入するか

**Reply by re-taro** [reply]
> - eslintのルール
- airbnbを入れる？
- Next.jsの公式のルールに則る
- @re-taro のカスタムルール

**Reply by re-taro** [reply]
> - components/下のディレクトリ構成
- Atomic Design
- なんちゃってAtomic Design
- Atomic Design 2.0
- なんか @re-taro がいい感じのアーキテクチャ(笑)を設計する

**Reply by re-taro** [reply]
> - Style lintのルール
- @Yusuke4869 が決めてくれ私はわからん

**Reply by re-taro** [reply]
> - commit lintを導入するか
- Conventional Commitsを強制する設定があったはず？(知らんけど)

**Reply by re-taro** [reply]
> - tsconfigの中身
- Next.jsの公式の設定でいいと思ってる
  - target: 'es6'より上にしたい

**Reply by re-taro** [reply]
> - Prettierのルール
- '' or ""
- ; or 
- , or

**Comment by re-taro** [comment]
# 新しい要件
- class名の命名規則が欲しい
- i18nどうしよう

**Reply by Yusuke4869** [reply]
class名は`oshavery-`から

**Reply by re-taro** [reply]
CIは回します
